ui/pages/notifications/notification-components/snap/snap.tsx

```diff
@@ -1,98 +1,161 @@
+} from '../../../../components/component-library';
+import { isOfTypeNodeGuard, type ExtractedNotification } from '../node-guard';
+import { SnapIcon } from '../../../../components/app/snaps/snap-icon';
 import { useMarkNotificationAsRead } from '../../../../hooks/metamask-notifications/useNotifications';
-import { useSnapNotificationTimeouts } from '../../../../hooks/useNotificationTimeouts';
-
-export type SnapNotification = Extract<
-  NotificationServicesController.Types.INotification,
-  { type: TRIGGER_TYPES.SNAP }
->;
-
-type SnapComponentProps = {
-  snapNotification: SnapNotification;
-};
-
-export const SnapComponent = ({ snapNotification }: SnapComponentProps) => {
-  const history = useHistory();
-  const trackEvent = useContext(MetaMetricsContext);
-  const { markNotificationAsRead } = useMarkNotificationAsRead();
-  const { setNotificationTimeout } = useSnapNotificationTimeouts();
 
-  const snapsMetadata = useSelector(getSnapsMetadata);
+type SnapNotification = ExtractedNotification<TRIGGER_TYPES.SNAP>;
 
-  const snapsNameGetter = getSnapName(snapsMetadata);
+type DetailedViewData = Extract<
+  SnapNotification['data'],
+  { origin: string; message: string; detailedView: ExpandedView }
+>;
 
-  const handleUnreadNotification = () => {
-    markNotificationAsRead([
-      {
-        id: snapNotification.id,
-        type: snapNotification.type,
-        isRead: snapNotification.isRead,
-      },
-    ]);
+export const components: NotificationComponent<SnapNotification> = {
+  guardFn: isOfTypeNodeGuard([TRIGGER_TYPES.SNAP]),
+  item: ({ notification, onClick }) => {
+    const history = useHistory();
+    const snapsMetadata = useSelector(getSnapsMetadata);
+    const snapsNameGetter = getSnapName(snapsMetadata);
+    const { markNotificationAsRead } = useMarkNotificationAsRead();
 
-    setNotificationTimeout(snapNotification.id);
-  };
+    const handleSnapButton = () => {
+      if (!notification.isRead) {
+        markNotificationAsRead([
+          {
+            id: notification.id,
+            type: notification.type,
+            isRead: notification.isRead,
+          },
+        ]);
+      }
 
-  const handleSnapClick = () => {
-    if (!snapNotification.isRead) {
-      handleUnreadNotification();
-    }
+      history.push(getSnapRoute(notification.data.origin));
+    };
 
-    trackEvent({
-      category: MetaMetricsEventCategory.NotificationInteraction,
-      event: MetaMetricsEventName.NotificationClicked,
-      properties: {
-        notification_id: snapNotification.id,
-        notification_type: snapNotification.type,
-        previously_read: snapNotification.isRead,
+    return (
+      <NotificationListItemSnap
+        id={notification.id}
+        snapId={notification.data.origin}
+        isRead={notification.isRead}
+        createdAt={new Date(notification.createdAt)}
+        title={{
+          items: [
+            {
+              text: snapsNameGetter(notification.data.origin) ?? 'Unknown Snap',
```

Comment:
Is the else case even possible here?

Reply:
Nah, remnants of old notifications team code. Removing it.